static/js/reducers/email.js

```diff
@@ -15,38 +16,57 @@ import {
 } from '../actions';
 import type { Action } from '../flow/reduxTypes';
 import type {
-  Email,
-  EmailState
+  AllEmailsState,
+  EmailState,
+  EmailInputs,
 } from '../flow/emailTypes';
 
 export const INITIAL_EMAIL_STATE: EmailState = {
-  email:  {},
+  inputs: {},
   validationErrors: {},
   sendError: {},
 };
+export const INITIAL_ALL_EMAILS_STATE: AllEmailsState = {
+  searchResultEmail: INITIAL_EMAIL_STATE,
+  courseTeamEmail: INITIAL_EMAIL_STATE
+};
 
-export const NEW_EMAIL_EDIT: Email = {
+export const NEW_EMAIL_EDIT: EmailInputs = {
   subject:    null,
   body:       null,
-  query:      null,
 };
 
-export const email = (state: EmailState = INITIAL_EMAIL_STATE, action: Action) => {
+function updatedState(state: AllEmailsState, emailType: string, updateObject: Object) {
+  _.merge(state[emailType], updateObject);
+  return _.cloneDeep(state);
```

Comment:
I did:

```
var _ = require('lodash')
let test = { inputType: { foo: 'bar', biz: 'boz'} }
_.merge(_.cloneDeep(test), {['inputType']: { foo: 'baz' }})
```

the `_.merge` line evaluated to:

```
{ inputType: { foo: 'baz', biz: 'boz' } };
```

so it seems like the `{ foo: 'baz' }` object was correctly merged with the `{ foo: 'bar', biz: 'boz' }` object.